wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+            return
+
+        logger.info(f"[Ban initban()] retrieved audit log data for ban actions: {audit_logs}")
+        logger.info(f"[Ban initban()] retrieved ban list from guild: {bans}")
+
+        logger.info("[Ban initban()] Starting process to move all guild bans into db")
+        ban_ids = [ban.user.id for ban in bans]
+        ban_data = []
+        audit_users = []
+        for ban in audit_logs:
+            audit_users.append(ban.target.id)
+
+            ban_data.append([ban.target.name+'#'+ban.target.discriminator,
+                             ban.target.id,
+                             ban.user.name+'#'+ban.user.discriminator,
+                             ban.user.id,
+                             pytz.utc.localize(ban.created_at),
+                             ban.reason if ban.reason else 'No Reason Given!'
+                             ])
+
+        for ban in bans:
+            if ban.user.id in audit_users:
+                continue
+            ban_data.append([ban.user.name+'#'+ban.user.discriminator,
+                             ban.user.id,
+                             None,
+                             None,
+                             None,
+                             'No Reason Given!'
+                             ])
+
+        # push to db and ban_list
+        for ban in ban_data:
+            if ban[1] in ban_ids:
+                self.ban_list.append(ban[1])
+                await BannedUsers.insert_ban(ban[0], ban[1])
+
+            await BanRecords.insert_record(ban[0], ban[1], ban[2], ban[3], ban[4], ban[5])
+        await ctx.send(f"{len(ban_data)} moved from guild bans to db.")
+        logger.info(f"[Ban initban()] total of {len(ban_data)} bans moved into db")
+
+    @commands.command()
+    async def ban(self, ctx, *args):
+        """Bans a user from the guild"""
+
+        logger.info(f"[Ban ban()] Ban command detected from {ctx.author} with args=[ {args} ]")
+        args = list(args)
+        mod_info = [ctx.author.name+'#'+ctx.author.discriminator, ctx.author.id]
+
+        # confirm at least 1 @ mention of user to ban
+        if len(ctx.message.mentions) < 1:
+            logger.info("[Ban ban()] No users were @ mentioned in the args")
+            e_obj = await em(ctx=ctx, title="Invalid Arguments",
+                             content=[("Error", "Please @ mention the user(s) to ban")],
+                             colour=self.errorColour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        # get mentions
+        users_to_ban = ctx.message.mentions
+        logger.info(f"[Ban ban()] list of user to be banned={users_to_ban}")
+
+        # construct reason, but first remove @'s from the args
+        reason = ''.join([i+' ' for i in args if not re.match(r'<@!?[0-9]*>', i)])
```

Comment:
instead of iterating through the args list and then doing a join, python doesn't offer a way to remove words from a string that match a certain pattern?

Reply:
i couldnt find shit

Reply:
this isn't helpful?
https://docs.python.org/3/library/re.html#re.sub